# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/QuicSocketControl.h
# Commit: 496ec0c5a60f
# Full Hash: 496ec0c5a60fdb86b233d036dd4747c477fc33ca
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2019-10-31 09:53:09
# Regressor Bug: 1581637
# File Overlap Count: 1
# Description:
#   Bug 1581637 - Part 8 - Add Http3Session/Http3Stream. r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46652
# ==============================================================================

diff -r 63b7f1ff1714 -r 496ec0c5a60f netwerk/protocol/http/QuicSocketControl.h
--- a/netwerk/protocol/http/QuicSocketControl.h	Wed Oct 30 20:54:21 2019 +0000
+++ b/netwerk/protocol/http/QuicSocketControl.h	Wed Oct 30 21:09:09 2019 +0000
@@ -12,8 +12,7 @@
 namespace mozilla {
 namespace net {
 
-// Will be added when Http3 lands.
-// class Http3Session;
+class Http3Session;
 
 // IID for the QuicSocketControl interface
 #define NS_QUICSOCKETCONTROL_IID                     \
@@ -37,8 +36,7 @@
   void SetInfo(uint16_t aCipherSuite, uint16_t aProtocolVersion,
                uint16_t aKeaGroup, uint16_t aSignatureScheme);
 
-  // Will be added when Http3 lands.
-  // void SetAuthenticationCallback(Http3Session *aHttp3Session);
+  void SetAuthenticationCallback(Http3Session* aHttp3Session);
   void CallAuthenticated();
 
   void HandshakeCompleted();